Improvements to sidecar injection doc (#9732)
istio/istio.io: content/en/docs/setup/additional-setup/sidecar-injection/index.md

In order to take advantage of all of Istio's features, pods in the mesh must be running an Istio sidecar proxy.

The following sections describe two
ways of injecting the Istio sidecar into a pod: enabling automatic Istio sidecar injection in the pod's namespace,
or by manually using the [`istioctl`](/docs/reference/commands/istioctl) command.

When enabled in a pod's namespace, automatic injection injects the proxy configuration at pod creation time using an admission controller.

> Manual injection directly modifies configuration, like deployments, and injects the proxy configuration into it.

Reviewer: ```suggestion
Manual injection directly modifies configuration, like deployments, by adding the proxy configuration into it.
```

Pods will, by default, use the `sidecar` injection template, which is automatically created.
This can be overridden by the `inject.istio.io/templates` annotation.
For example, to apply the default template and our customization, you can set `inject.istio.io/templates=sidecar,custom`.

> In addition to the `sidecar`, a `gateway` template is provided by default to allow injection of Gateway deployments.

Reviewer: ```suggestion
In addition to the `sidecar`, a `gateway` template is provided by default to support proxy injection into Gateway deployments.
```
